Engine/Common/Types.h

```diff
@@ -431,199 +699,271 @@ struct Float2 {
     {
-        return X >= other.X && Y >= other.Y;
-    };
-    DLLEXPORT inline bool operator==(const Float2& other) const
+        return std::tie(X, Y) >= std::tie(other.X, other.Y);
+    }
+
+    DLLEXPORT inline bool operator==(const Float2& other) const noexcept
     {
         return X == other.X && Y == other.Y;
-    };
-    DLLEXPORT inline bool operator!=(const Float2& other) const
+    }
+
+    DLLEXPORT inline bool operator!=(const Float2& other) const noexcept
     {
-        return X != other.X && Y != other.Y;
-    };
+        return X != other.X || Y != other.Y;
+    }
+
     // ------------------ Functions ------------------ //
-    DLLEXPORT inline float GetX() const
+    // getters //
+    DLLEXPORT inline float GetX() const noexcept
     {
         return X;
     }
-    DLLEXPORT inline float GetY() const
+
+    DLLEXPORT inline float GetY() const noexcept
     {
         return Y;
     }
-    DLLEXPORT inline void SetX(const float& val)
+
+    // setters //
+    DLLEXPORT inline void SetX(float val)
     {
         X = val;
         DO_NAN_CHECK;
-    };
-    DLLEXPORT inline void SetY(const float& val)
+    }
+
+    DLLEXPORT inline void SetY(float val)
     {
         Y = val;
         DO_NAN_CHECK;
-    };
+    }
 
     // add all elements together //
-    DLLEXPORT inline float HAdd() const
+    DLLEXPORT inline float HAdd() const noexcept
     {
         return X + Y;
     }
+
     // Add all elements together after abs() is called on each element //
-    DLLEXPORT inline float HAddAbs() const
+    DLLEXPORT inline float HAddAbs() const noexcept
     {
         return std::fabs(X) + std::fabs(Y);
     }
+
     // getting min and max of objects //
-    DLLEXPORT inline Float2 MinElements(const Float2& other) const
+    DLLEXPORT inline Float2 MinElements(const Float2& other) const noexcept
     {
         return Float2(X < other.X ? X : other.X, Y < other.Y ? Y : other.Y);
     }
-    DLLEXPORT inline Float2 MaxElements(const Float2& other) const
+
+    DLLEXPORT inline Float2 MaxElements(const Float2& other) const noexcept
     {
         return Float2(X > other.X ? X : other.X, Y > other.Y ? Y : other.Y);
     }
+
     // value clamping //
-    DLLEXPORT inline Float2 Clamp(const Float2& min, const Float2& max)
+    DLLEXPORT inline Float2 Clamp(const Float2& min, const Float2& max) const noexcept
     {
         const Float2 minval = this->MinElements(max);
         return min.MaxElements(minval);
     }
 
     // ----------------- Vector math ------------------- //
     // dot product of the vectors //
-    DLLEXPORT inline float Dot(const Float2& val) const
+    DLLEXPORT inline float Dot(const Float2& val) const noexcept
     {
         return X * val.X + Y * val.Y;
     }
+
     // length of the vector //
-    DLLEXPORT inline float Length() const
+    DLLEXPORT inline float Length() const noexcept
     {
-        return sqrt(X * X + Y * Y);
+        return static_cast<float>(sqrt(X * X + Y * Y));
```

Comment:
I think `std::sqrt` should be used instead of the cast here. Maybe removing `cmath` header would get rid of the C-style math functions to clean things up?